src/Spaces/box.jl

```diff
@@ -74,8 +74,8 @@ function contains(x, box_obj::Box)
 end
 
 checkvalidtypes(box_obj1::Box, box_obj2::Box) =
-    (box_obj1.dtype <: Integer && box_obj2.dtype <: Integer) ?   # If both dtypes are Integers...
-        (box_obj1.dtype <: Unsigned && box_obj2.dtype <: Unsigned) || (box_obj1.dtype <: Signed && box_obj2.dtype <: Signed) :  # Check if they're both Signed or both Unsigned...
-        (box_obj1.dtype <: AbstractFloat && box_obj2.dtype <: AbstractFloat)  # Otherwise, check if they're both Floats.
+    box_obj1.dtype == box_obj2.dtype ||
+        box_obj1.dtype <: Integer && box_obj2.dtype <: Integer &&
+            (box_obj1.dtype <: Unsigned && box_obj2.dtype <: Unsigned) || (box_obj1.dtype <: Signed && box_obj2.dtype <: Signed)
```

Comment:
We can only check for the `dtype` to be (Un)signed, since it is sub-type of Integer.